### PR TITLE
chore: consolidate tree view

### PR DESCRIPTION
**Pull request by Harshita-mindfire**
This PR aims to consolidate code for Tree View between Local and Web Versions. More details here: [[Consolidate Tree View Code|dendron://private/task.2022.09.01.consolidate-tree-view-code]]

# Dendron Extended PR Checklist

- NOTE: the links don't work. you'll need to go into the wiki and use lookup to find the note until we fix some issues in the markdown export

### Basics

- [x] code should follow [Code Conventions](dev.process.code)
- [x] circular dependency check: make sure your code is not introducing new circular dependencies in plugin-core.  See [Avoiding Circular Dependencies](dev.process.code.best-practices).
- [ ] sticking to existing conventions instead of creating new ones
  - eg: [if configuration for utilities are already in one module or package, add future utilities there as well](https://github.com/dendronhq/dendron/pull/1960#discussion_r786228021)

### Extended
- General
  - [ ] check whether code be simplified
  - [x] check if similar function already exist in the codebase. if so, can it be re-used?
  - [ ] check if this change adversely impact performance
- Operations
  - [ ] when shipping this change, will it just work or will it introduce additional operational overhead due to complicated interface or known bugs?
- Architecture
  - [ ] check if code is introducing changes on a foundational class or interface. if so, call for design review if needed
    - eg: [making changes to DNode](https://github.com/dendronhq/dendron/pull/2158#pullrequestreview-854689586)

### Basics
- [ ] if you are adding analytics related changes, make sure the [Telemetry](https://wiki.dendron.so/notes/84df871b-9442-42fd-b4c3-0024e35b5f3c.html) docs are updated

### Extended
- [~] can we track the performance of this change to know if it is _successful_?
  - eg: [see usage for export pod](https://github.com/dendronhq/dendron/pull/2190#pullrequestreview-855715612)

### Basics

- [x] [Write Tests](dev.process.qa.test) 
- [x] [Confirm existing tests pass](dev.process.qa.test)
- [x] [Confirm manual testing](dev.process.qa.test) 
- [x] Common cases tested
- [ ] 1-2 Edge cases tested
- [ ] If your tests changes an existing snapshot, snapshots have been [updated](dev.process.qa.test)

## Docs
- [~] if your change reflects documentation changes, also submit a PR to [dendron-site](https://github.com/dendronhq/dendron-site) and mention the doc PR link in your current PR
- [ ] does this change introduce a new or better way of doing things that others need to be aware of? if so, an async should be created and a process added in [Development](dev) or [Packages](pkg)

### Extended
- [~]  is this a developer BREAKING change? if another person cloning from this branch will need to adjust their dependencies or mental model of the architecture, then it is. if this is the case, make sure this is communicated according to [Close Loop](dev.process.close-loop)
  - eg: [breaking dev change due to new dependency](https://github.com/dendronhq/dendron/pull/2188#pullrequestreview-855696330)

### CODE REVIEW

File: packages/plugin-core/src/common/EngineNoteProvider.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import "reflect-metadata";
```
Comment (jonathanyeung) [inline packages/plugin-core/src/common/EngineNoteProvider.ts:1]: was this one needed for a test to work? 

If so, I think we can configure this in Jest (lemme dig up some code for you)

**Review by jonathanyeung — COMMENTED** [review]
looking good!

File: packages/plugin-core/src/common/EngineNoteProvider.ts (hunk 1 above)
Comment (Harshita-mindfire) [reply packages/plugin-core/src/common/EngineNoteProvider.ts]: Na, this import isn't required. I might have added accidently.

**Comment by Harshita-mindfire** [comment]
loom: https://www.loom.com/share/0d1b01b1c3654c75b97e40c1aa8d5f76

File: packages/plugin-core/src/commands/CreateNoteCommand.ts [hunk 2]
```diff
@@ -8,10 +8,10 @@ import {
 import { AutoCompletableRegistrar } from "../utils/registers/AutoCompletableRegistrar";
 import _ from "lodash";
 import { window } from "vscode";
-import { NoteProps, NoteUtils } from "@dendronhq/common-all";
 import { AnalyticsUtils } from "../utils/analytics";
+import { ExtensionProvider } from "../ExtensionProvider";
 
-type CommandOpts = NoteProps & {};
+type CommandOpts = string & {};
```
Comment (namjul) [inline packages/plugin-core/src/commands/CreateNoteCommand.ts:14]: The suffix `Opts` means to me option-object than a scalar value. Maybe we should keep it being an option-object?

File: packages/plugin-core/src/injection-providers/setupLocalExtContainer.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  const { wsRoot, engine, vaults } = opts;
+  container.register<EngineEventEmitter>("EngineEventEmitter", {
+    useToken: "ReducedDEngine",
+  });
+  container.register("wsRoot", { useValue: vscode.Uri.file(wsRoot) });
+  container.register("ReducedDEngine", { useValue: engine });
+  container.register("vaults", { useValue: vaults });
+  container.register<ITreeViewConfig>("ITreeViewConfig", {
+    useClass: TreeViewDummyConfig,
+  });
```
[the remark this answers is not in the document]
Comment (Harshita-mindfire) [reply packages/plugin-core/src/injection-providers/setupLocalExtContainer.ts:23]: the injection provider for `web` is done with `setupWebExtContainer` and incase of web, this won't run.

[the remark this answers is not in the document]
Comment (hikchoi) [reply packages/plugin-core/src/injection-providers/setupLocalExtContainer.ts:23]: I'm aware of that. But wouldnt using the dummy config in local stop it from persisting in local as well?

[the remark this answers is not in the document]
Comment (Harshita-mindfire) [reply packages/plugin-core/src/injection-providers/setupLocalExtContainer.ts:23]: Oh, I got what you are saying now. You mean `"treeViewItemLabelType": "filename"` config from meta.json right?

[the remark this answers is not in the document]
Comment (jonathanyeung) [reply packages/plugin-core/src/injection-providers/setupLocalExtContainer.ts:23]: Yeah, good catch @hikchoi  - what we'll need to do here is create a new impl of `ITreeViewConfig` that uses the metadata service.

[the remark this answers is not in the document]
Comment (Harshita-mindfire) [reply packages/plugin-core/src/injection-providers/setupLocalExtContainer.ts]: [Commit](https://github.com/dendronhq/dendron/pull/3499/commits/2ad81eb97c6d2572edf1e278e85fe6567405e8ec)

File: packages/plugin-core/src/common/EngineNoteProvider.ts [hunk 4]
```diff
@@ -19,7 +19,7 @@ import {
   TreeItemCollapsibleState,
```
Comment (jonathanyeung) [inline packages/plugin-core/src/common/EngineNoteProvider.ts:19]: Can you move all the files that are a part of this review from this location to `src\views\common\treeview` ?

File: packages/plugin-core/src/views/TreeViewConfig.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import { TreeViewItemLabelTypeEnum } from "@dendronhq/common-all";
+import { MetadataService } from "@dendronhq/engine-server";
+import { ITreeViewConfig } from "../common/ITreeViewConfig";
+
+/**
+ * Config for Tree View when extension is run locally- this version pull values from
+ * MetadataService
+ */
+export class TreeViewConfig implements ITreeViewConfig {
```
Comment (jonathanyeung) [inline packages/plugin-core/src/views/TreeViewConfig.ts:9]: Can we move this to `src\views\node\treeview` and then rename the file/class to `MetadataSvcTreeViewConfig`?

**Review by jonathanyeung — APPROVED** [review]
Two minor comments about file positions, otherwise lgtm to merge!

**Review by hikchoi — APPROVED** [review]

**Pull request merged**